bindings/cs/MongoDB.Libmongocrypt/LibraryLoader.cs

```diff
@@ -58,42 +58,46 @@ public LibraryLoader()
+            switch (OperatingSystemHelper.CurrentOperatingSystem)
             {
-                string[] suffixPaths = new[]
-                {
-                    "../../native/osx/",
-                    string.Empty
-                };
-                string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.dylib");
-                _loader = new DarwinLibraryLoader(path);
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-            {
-                string[] suffixPaths = new[]
-                {
-                    "../../native/linux/",
-                    string.Empty
-                };
-                string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.so");
-                _loader = new LinuxLibrary(path);
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-
-#endif
-            {
-                string[] suffixPaths = new[]
-                {
-                    @"..\..\x64\native\windows\",
-                    string.Empty
-                };
-                string path = FindLibrary(candidatePaths, suffixPaths, "mongocrypt.dll");
-                _loader = new WindowsLibrary(path);
+                case OperatingSystemPlatform.MacOS:
+                    {
+                        string[] suffixPaths = new[]
+                        {
+                            "../../native/osx/",
+                            string.Empty
+                        };
+                        string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.dylib");
+                        _loader = new DarwinLibraryLoader(path);
+                    }
+                    break;
+                case OperatingSystemPlatform.Linux:
+                    {
+                        string[] suffixPaths = new[]
+                        {
+                            "../../native/linux/",
+                            string.Empty
+                        };
+                        string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.so");
+                        _loader = new LinuxLibrary(path);
+                    }
+                    break;
+                case OperatingSystemPlatform.Windows:
+                    {
+                        string[] suffixPaths = new[]
+                        {
+                            @"..\..\x64\native\windows\",
+                            string.Empty
+                        };
+                        string path = FindLibrary(candidatePaths, suffixPaths, "mongocrypt.dll");
+                        _loader = new WindowsLibrary(path);
+                    }
+                    break;
+                default:
+                    // should not be reached. If we're here, then there is a bug in OperatingSystemHelper
+                    throw new PlatformNotSupportedException("Unexpected operating system.");
```

Comment:
Nit: the leftover is `FreeBSD` ([OSPlatform](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.osplatform?view=netcore-3.1)). Not sure if we want to handle it in some specific way.

Reply:
As far as I know, we won't support it